src/Orleans/Serialization/FieldUtils.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace Orleans.Serialization
+{
+    public sealed class FieldUtils
+    {
+        /// <summary>
+        /// The delegate used to set fields in value types.
+        /// </summary>
+        /// <typeparam name="TDeclaring">The declaring type of the field.</typeparam>
+        /// <typeparam name="TField">The field type.</typeparam>
+        /// <param name="instance">The instance having its field set.</param>
+        /// <param name="value">The value being set.</param>
+        public delegate void ValueTypeSetter<TDeclaring, in TField>(ref TDeclaring instance, TField value);
```

Comment:
I believe @ReubenBond meant to move it to this code file, not necessarily as a nested type. I'm fine merging it regardless, but I also feel it makes more sense to move it outside the class, inside the same codefile.

Reply:
This is fine by me